File: backend/src/routes/integrations.routes.ts
import { Router } from 'express'
import { requireAuth } from '../middlewares/auth.middleware'
import {
  integrationStatus,
  googleConnect, googleCallback, googleDisconnect, googleListEvents, googleCreateEvent,
  outlookConnect, outlookCallback, outlookDisconnect, outlookCreateEvent,
} from '../controllers/integrations.controller'

const router = Router()

// Status de conexão (requer auth)
router.get('/status', requireAuth, integrationStatus)

// Google Calendar
router.get('/google/connect',  requireAuth, googleConnect)
router.get('/google/callback', googleCallback)
router.delete('/google',       requireAuth, googleDisconnect)
router.get('/google/events',   requireAuth, googleListEvents)
router.post('/google/events',  requireAuth, googleCreateEvent)

// Outlook
router.get('/outlook/connect',  requireAuth, outlookConnect)
router.get('/outlook/callback', outlookCallback)         // sem auth — callback público do OAuth
router.delete('/outlook',       requireAuth, outlookDisconnect)
router.post('/outlook/events',  requireAuth, outlookCreateEvent)

export default router

File: backend/internal/integrations/integrations_routes.go
package integrations

import (
	"github.com/gin-gonic/gin"
	"github.com/jackc/pgx/v5/pgxpool"

	"beeplanerofc/backend-go/internal/config"
	"beeplanerofc/backend-go/internal/middleware"
	"beeplanerofc/backend-go/internal/supabaseauth"
)

// RegisterRoutes registra as rotas de integrações no grupo /api/integrations.
// Os callbacks OAuth (google/callback, outlook/callback) são públicos — o provedor
// redireciona pra lá sem token, a identidade do usuário vem do "state" codificado.
func RegisterRoutes(router *gin.RouterGroup, db *pgxpool.Pool, authClient *supabaseauth.Client, env *config.Env) {
	auth := middleware.RequireAuth(db, authClient)

	status := NewStatusController(db)
	google := NewGoogleController(db, env)
	outlook := NewOutlookController(db, env)

	router.GET("/status", auth, status.Status)

	router.GET("/google/connect", auth, google.Connect)
	router.GET("/google/callback", google.Callback)
	router.DELETE("/google", auth, google.Disconnect)
	router.GET("/google/events", auth, google.ListEvents)
	router.POST("/google/events", auth, google.CreateEvent)

	router.GET("/outlook/connect", auth, outlook.Connect)
	router.GET("/outlook/callback", outlook.Callback)
	router.DELETE("/outlook", auth, outlook.Disconnect)
	router.POST("/outlook/events", auth, outlook.CreateEvent)
}
